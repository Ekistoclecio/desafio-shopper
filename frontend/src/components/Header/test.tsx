import { screen } from '@testing-library/react';

import { Header } from '.';

import { renderWithTheme } from '@/utils/tests/helpers';

describe('<Header />', () => {
  it('should render the Header', () => {
    const { container } = renderWithTheme(<Header title="home" />);

    expect(screen.getByRole('heading', { name: /home/i })).toBeInTheDocument();

    const links = screen.getAllByRole('link');

    expect(links).toHaveLength(2);

    expect(links[0]).toHaveAttribute('href', '/');
    expect(links[1]).toHaveAttribute('href', '/history');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Header without links', () => {
    renderWithTheme(<Header title="home" hiddenNav />);

    expect(screen.queryAllByRole('link')).toHaveLength(0);
  });
});
