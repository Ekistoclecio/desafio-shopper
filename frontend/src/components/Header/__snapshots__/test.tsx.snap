// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the Header 1`] = `
<div
  class="MuiBox-root css-1dvu9bo"
>
  <div
    class="MuiBox-root css-rozte9"
  >
    <div
      class="MuiBox-root css-f5t1ey"
    />
    <h4
      class="MuiTypography-root MuiTypography-h4 css-6041mz-MuiTypography-root"
    >
      home
    </h4>
  </div>
  <div
    class="MuiBox-root css-yzg3r8"
  >
    <a
      aria-current="page"
      class="css-i844pp active"
      data-discover="true"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="HomeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </a>
    <a
      class="css-i844pp"
      data-discover="true"
      href="/history"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="HistoryIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9m-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8z"
        />
      </svg>
    </a>
  </div>
</div>
`;
